sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -4600,3 +4600,40 @@ case class ArrayExcept(left: Expression, right: Expression) extends ArrayBinaryL
   override protected def withNewChildrenInternal(
     newLeft: Expression, newRight: Expression): ArrayExcept = copy(left = newLeft, right = newRight)
 }
+
+@ExpressionDescription(
+  usage = "_FUNC_(array) - Removes null values from the array.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(array(1, 2, 3, null));
+       [1,2,3]
+  """,
+  group = "array_funcs",
+  since = "3.4.0")
+case class ArrayCompact(expr: Expression, replacement: Expression)
+  extends RuntimeReplaceable with InheritAnalysisRules {
+
+  def this(expr: Expression) = this(expr,
+    ArrayFilter(expr, ArrayCompact.createLambdaExpr()))
+
+  override def parameters: Seq[Expression] = Seq.empty
+
+  override def prettyName: String = "array_compact"
+
+  override protected def withNewChildInternal(newChild: Expression): ArrayCompact =
+    copy(replacement = newChild)
+}
+
+object ArrayCompact {
+
+  def apply(expr: Expression): ArrayCompact = {
```

Comment:
Do we really need `apply` ?

Reply:
Yes, because in `functions.scala`  `ArrayCompact` is created without new `operator`. If required I can add new `operator` and remove apply from here, WDYT ?

```
  def array_compact(column: Column): Column = withExpr {
    ArrayCompact(column.expr)
  }
```
to 

```
  def array_compact(column: Column): Column = withExpr {
    new ArrayCompact(column.expr)
  }
```

Reply:
Updated, Thank you